n2v/methods/wuyang.py

```diff
@@ -129,18 +152,135 @@ def hessian(self, v):
         Ha = 2 * contract('iau,iat,ia->ut', C3a, C3a, eigs_diff_a**-1)
 
         if self. ref == 1:
+            if self.lambda_reg is not None:
+                Ha += 4 * self.T_pbs
             Hs = Ha
 
         else:
             eigs_diff_b = self.eigvecs_b[:nb, None] - self.eigvecs_b[None, nb:]
             C3b = contract('mi,va,mvt->iat', self.Cb[:,:nb], self.Cb[:,nb:], self.S3)
             Hb = 2 * contract('iau,iat,ia->ut', C3b, C3b, eigs_diff_b**-1)
-            Hs = np.block( 
-                            [[Ha,                               np.zeros((self.naux, self.naux))],
-                            [np.zeros((self.naux, self.naux)), Hb                              ]] 
+            if self.lambda_reg is not None:
+                Hb += 4 * self.T_pbs
+            Hs = np.block(
+                            [[Ha,                               np.zeros((self.npbs, self.npbs))],
+                            [np.zeros((self.npbs, self.npbs)), Hb                              ]]
                         )
 
-        if self.reg > 0.0:
-            pass
-
         return - Hs
+
+    def find_regularization_constant(self, opt_method, potential_components, tol=None, opt=None, lambda_list=None):
+        """
+        Finding regularization constant lambda.
+
+        Note: it is recommend to set a specific convergence criteria by opt or tol,
+                in order to control the same convergence
+                for different lambda value.
+
+        After the calculation is done, one can plot the returns to select a good lambda.
+
+        Parameters:
+        -----------
+        opt_method: string
+            opt_methods available in scipy.optimize.minimize
+
+        potential_components: the components for guide potential v0.
+
+        tol: float
+            Tolerance for termination. See scipy.optimize.minimize for details.
+
+        opt: default None, or dictionary
```

Comment:
The PEP docstring convention suggests adding optional arguments the same and then just specify that is optional, e.g.:
```
opt_method: str, optional  
       Type of solver...
```
So I believe we can skip the "If None" and just explain what happens if it's there.